applications/matter_bridge/src/bridged_device_types/ble_onoff_light_data_provider.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "ble_bridged_device.h"
+#include "ble_connectivity_manager.h"
+#include "bridged_device_data_provider.h"
+
+#include <bluetooth/services/lbs.h>
+
+class BleOnOffLightDataProvider : public BLEBridgedDeviceProvider {
+public:
+	BleOnOffLightDataProvider(UpdateAttributeCallback callback) : BLEBridgedDeviceProvider(callback) {}
+	~BleOnOffLightDataProvider() {}
+
+	void Init() override;
+	void NotifyUpdateState(chip::ClusterId clusterId, chip::AttributeId attributeId, void *data,
+			       size_t dataSize) override;
+	CHIP_ERROR UpdateState(chip::ClusterId clusterId, chip::AttributeId attributeId, uint8_t *buffer) override;
+
+	static void NotifyAttributeChange(intptr_t context);
+	static void GattWriteCallback(struct bt_conn *conn, uint8_t err, struct bt_gatt_write_params *params);
+	static uint8_t GattNotifyCallback(struct bt_conn *conn, struct bt_gatt_subscribe_params *params,
+					  const void *data, uint16_t length);
+
+	bt_uuid *GetServiceUuid() override;
+	void MatchBleDevice(BLEBridgedDevice *device) override
+	{
+		mDevice = device;
```

Comment:
Yeah, it was actually checked before invoking this method, but still it would be better to have a check inside as well. I changed method API to return error code and validate pointer.